elastic/kibana
# [Infra] Handle view in app for legacy metrics

closes [#189625](https://github.com/elastic/kibana/issues/189625)

## Summary

This PR changes the asset details to display a call if the user comes from the alerts page via an inventory rule created with one of the legacy metrics.

Besides that, it changes how the link is built to use locators.

Legacy metrics example

https://github.com/user-attachments/assets/12308f4e-e269-4580-b86d-808ae9f6fe10

**Regression**

Metrics Threshold

https://github.com/user-attachments/assets/94032f51-6b2c-4760-8019-158746a1aa13

Inventory Rule (new/hosts view metrics)

https://github.com/user-attachments/assets/0f872f3a-7bdb-4fb8-a925-7ed3621fee2d

Inventory Rule (custom metric)


https://github.com/user-attachments/assets/f2e5ded5-b2e6-45ff-878d-6361c4540140

### Fix

While working on it, I discovered that alerts for containers were not redirecting the users to the asset details page for containers. That was fixed too

Inventory rule for containers

https://github.com/user-attachments/assets/05f20c12-6fdc-45c0-bc38-b756bfbf3658

Metrics threshold rule for containers


### How to test

- Start a local Kibana instance (easier if pointed to an oblt cluster)
- Create Inventory Rule alerts for:  
  - host: 1 legacy metric and 1 non-legacy metric
  - container
- Create Metric Threshold alerts with
  - avg on `system.cpu.total.norm.pct` grouped by `host.name` 
  - avg on `kubernetes.container.cpu.usage.limit.pct` grouped by `container.id`
- Navigate to the alerts page and click on the `view in app` button, as shown in the recordings above
- Test if the navigation to the asset details page works
   - For a legacy metric, the callout should be displayed
   - Once dismissed, the callout should not appear again for that metric
